test/integration_tests/python/test_jobs.py

```diff
@@ -94,3 +94,5 @@ def test_jobs(with_hierarchy, with_gear, with_invalid_gear, as_user, as_admin):
 
     r = as_user.post('/jobs/' + job1_id + '/retry')
     assert r.ok
+
+    db.jobs.remove()
```

Comment:
This is probably a dumb comment, but - this ingrains a pretty strong assumption that we won't be running unit tests in parallel. I'm guessing there are assumptions like that all over the tests; curious what your thoughts are.